src/views/Graph.vue

```diff
@@ -86,6 +86,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
             />
           </g>
         </g>
+        <g v-if="groupCycle" class="graph-subgraph-container">
+          <GraphSubgraph
+            v-for="(subgraph, key) in subgraphs"
```

Comment:
We may be able to move that `-vif` out of the `GraphSubgraph` component to avoid creating component instances for the renderless "hidden" subgraphs.

```js
          <GraphSubgraph
            v-for="(subgraph, key) in subgraphs"
            v-if="subgraph.label!='margin'"
```

Reply:
Cant have v-for and v-if on the same element but think another solution is

```        
        <g v-if="groupCycle" class="graph-subgraph-container">
          <g v-for="(subgraph, key) in subgraphs"
            :key="key">
            <GraphSubgraph
              v-if="subgraph.label!='margin'"
              :subgraph="subgraph" />
          </g>
        </g>
```

Reply:
This would fail ESLint - https://eslint.vuejs.org/rules/no-use-v-if-with-v-for.html

Reply:
What is the purpose of the `v-if`? I saw in an earlier discussion https://github.com/cylc/cylc-ui/pull/1763#discussion_r1587800644

```
:visibility="subgraph.label=='margin' ?  'hidden' : 'visible' "
```
but `v-if` does not accomplish the same thing (hidden visibility means it still takes up space which I presume is the whole point)

Reply:
These "hidden" subgraphs are added into the GraphViz document in order to take up space in the virtual graph (relative coordinates) proving a bit of margin between subgraphs which is nice.

However, I don't think they need to be added into the SVG document (either as hidden elements or otherwise). You don't need them to take up space in the SVG (absolute coordinates), we only need the x,y,width,height of the subgraph outline.

Another way of doing it would be to filter out the "hidden" clusters from `this.subgraphs`. From an efficiency standpoint it's best to reduce the number of HTML elements and especially Vue components we use.

Reply:
No - the layout is done by graphviz and what we are rendering in the svg is a copy of that. The spacing between graphviz subgraphs is a bit random and so I added an extra subgraph to give a handle for controlling the 'margin' between the different subgraphs (or at least give a minimum spacing).
This means each cycle point has two subgraphs - one that it is meant to have and a second which is only there to control the spacing...
![image](https://github.com/cylc/cylc-ui/assets/7523921/6e08e812-5c45-4387-be90-68fff84651f4)

So just need to make that "margin" one not visible/render.

Reply:
So the subgraph for the margin/spacing is needed in the dot code but not in the rendered DOM?